connect_test.go

```diff
@@ -76,68 +80,136 @@ func Test_HCPConnect_FlagValidation(t *testing.T) {
+		getCallerIdentityErr error
+		expectedResult int
+	}{
+		"OK resp": {
+			getCallerIdentityResp: &hcpis.IamServiceGetCallerIdentityOK{
+				Payload: &iam_models.HashicorpCloudIamGetCallerIdentityResponse{
+					Principal: &iam_models.HashicorpCloudIamPrincipal{
+						User: &iam_models.HashicorpCloudIamUserPrincipal{
+							Email:    "test@test.com",
+							FullName: "HCP Test",
+							ID:       "test",
+							Subject:  "test",
+						},
 					},
 				},
 			},
-		}, nil)
-
-	mockRmProjClient := projmocks.NewClientService(t)
-	mockRmProjClient.
-		On("ProjectServiceList", mock.Anything, nil).
-		Return(&hcprmp.ProjectServiceListOK{
-			Payload: &models.HashicorpCloudResourcemanagerProjectListResponse{
-				Projects: []*models.HashicorpCloudResourcemanagerProject{
-					{
-						ID:   uuid.New().String(),
-						Name: "mock-project-1",
-						State: models.NewHashicorpCloudResourcemanagerProjectProjectState(
-							models.HashicorpCloudResourcemanagerProjectProjectStateACTIVE,
-						),
-					},
-				},
+			getCallerIdentityErr: nil,
+			expectedResult: 0,
+		},
+		"no resp or error": {
+			getCallerIdentityResp: nil,
+			getCallerIdentityErr: nil,
+			expectedResult: 0,
+		},
+		"error - unauthorized": {
+			getCallerIdentityResp: nil,
+			getCallerIdentityErr: hcpis.NewIamServiceGetCallerIdentityDefault(http.StatusUnauthorized),
+			expectedResult: 0,
+		},
+		"error - server error": {
+			getCallerIdentityResp: nil,
+			getCallerIdentityErr: hcpis.NewIamServiceGetCallerIdentityDefault(http.StatusInternalServerError),
+			expectedResult: 1,
+		},
+		"nil payload": {
+			getCallerIdentityResp: &hcpis.IamServiceGetCallerIdentityOK{
+				Payload: nil,
 			},
-		}, nil)
-
-	mockVsClient := clustermocks.NewClientService(t)
-	mockVsClient.
-		On("Get", mock.Anything, nil).
-		Return(&hcpvs.GetOK{
-			Payload: &hcpvsm.HashicorpCloudVault20201125GetResponse{
-				Cluster: &hcpvsm.HashicorpCloudVault20201125Cluster{
-					ID:       "cluster-1",
-					DNSNames: &hcpvsm.HashicorpCloudVault20201125ClusterDNSNames{Proxy: "hcp-proxy-cluster-1.addr:8200"},
-					State: hcpvsm.NewHashicorpCloudVault20201125ClusterState(
-						hcpvsm.HashicorpCloudVault20201125ClusterStateRUNNING,
-					),
-					Config: &hcpvsm.HashicorpCloudVault20201125ClusterConfig{
-						NetworkConfig: &hcpvsm.HashicorpCloudVault20201125NetworkConfig{
-							HTTPProxyOption: hcpvsm.NewHashicorpCloudVault20201125HTTPProxyOption(hcpvsm.HashicorpCloudVault20201125HTTPProxyOptionENABLED),
-						},
-					},
+			getCallerIdentityErr: nil,
+			expectedResult: 0,
+		},
+		"nil principal": {
+			getCallerIdentityResp: &hcpis.IamServiceGetCallerIdentityOK{
+				Payload: &iam_models.HashicorpCloudIamGetCallerIdentityResponse{
+					Principal: nil,
 				},
 			},
-		}, nil)
+			getCallerIdentityErr: nil,
+			expectedResult: 0,
+		},
+	}
+
+	for name, test := range tests {
+		t.Run(name, func(t *testing.T) {
+			_, cmd := testHCPConnectCommand()
```

Comment:
Is there nothing we can validate on the `ui` that gets returned here? Even just 'is it non-nil' could be useful, I think

Reply:
Great question. Probably! I modified this test to be table-driven so the assertions are essentially the same as from before. I think validating the output likely makes sense. Will give it a shot.

Reply:
Ah, definitely no worries! I would have considered that 'outside the scope of review' normally, but if you want to give it a go, it does seem like it might help!